app/src/main/java/com/github/se/eventradar/ui/login/SignUpScreen.kt

```diff
@@ -0,0 +1,469 @@
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+// import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import coil.compose.rememberImagePainter
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.github.se.eventradar.R
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import java.text.SimpleDateFormat
+
+// private const val TAG = "SignUpScreen"
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SignUpScreen(navigationActions: NavigationActions) {
+
+    val openErrorDialog = remember { mutableStateOf(false) }
+
+    // val context = LocalContext.current
+    val selectedImageUri = rememberSaveable { mutableStateOf<Uri?>(null) }
```

Comment:
nit: Just a reminder we need to create validation checks so that if user doesn't enter a field/enters the wrong format they get an error message but can be implemented later.

Reply:
Normally, all fields have checks except username as it will be taken care of with the implementation of the storing of user data:
- Username: to be implemented in other PR
- Name: should not be empty
- Surname: should not be empty
- Phone Number: should be of the correct length
- Birth Date: should be a date in the format (DD/MM/YYYY)